### PR TITLE
Improve the property names of the transformer classes

### PR DESCRIPTION
**Pull request by Menrath**
I tried to improve the readability and the efficiency of certain custom transformers via:

- Renaming the property 'object' to 'wp_object' of the _Transformer_ classes  to indicate that its the source
- Adding an optional property 'activitypub_object' which may make already transformed parts available to other "getter" functions.
- Enhancing some PHP-Docs

### CODE REVIEW

File: includes/transformer/class-base.php [hunk 1]
```diff
@@ -46,9 +61,9 @@ public function __construct( $object ) {
 	 * @return Activitypub\Activity\Base_Object
 	 */
 	public function to_object() {
-		$object = new Base_Object();
+		$this->activitypub_object = new Base_Object();
 
-		$vars = $object->get_object_var_keys();
+		$vars = $this->activitypub_object->get_object_var_keys();
```
Comment (pfefferle) [inline includes/transformer/class-base.php:66]: why storing it in a local var?

Comment (pfefferle) [reply includes/transformer/class-base.php:66]: it is not used outside of this function, so it might be more confusing than it would help?!?

Comment (Menrath) [reply includes/transformer/class-base.php:66]: You are right. From this Pull Requests point of view custom transformers might override the to_object function anyway. And they could also define their own protected property for storing the already processed part of the object. 

I would suggest reverting this part of the code and deal with the performance optimizations by reusing already transformed parts later. One would also then have to take care of the race-conditions which properties gets transformed when.

File: includes/transformer/class-base.php [hunk 2]
```diff
@@ -12,18 +15,20 @@
...
-	protected $object;
+	protected $wp_object;
 
 	/**
 	 * Static function to Transform a WordPress Object.
 	 *
 	 * This helps to chain the output of the Transformer.
 	 *
-	 * @param stdClass $object The WP_Post object
+	 * @param WP_Object|WP_Comment $object The WP_Post object
```
Comment (pfefferle) [inline includes/transformer/class-base.php:31]:
```suggestion
	 * @param stdClass|WP_Post|WP_Comment $wp_object The WordPress object
```

File: includes/transformer/class-base.php [hunk 3]
```diff
@@ -34,10 +39,10 @@ public static function transform( $object ) {
 	/**
 	 * Base constructor.
 	 *
-	 * @param stdClass $object
+	 * @param WP_Object|WP_Comment $wp_object
```
Comment (pfefferle) [inline includes/transformer/class-base.php:42]:
```suggestion
	 * @param stdClass|WP_Post|WP_Comment $wp_object The WordPress object
```

File: includes/transformer/class-base.php [hunk 4]
```diff
@@ -12,18 +15,20 @@
  */
 abstract class Base {
 	/**
-	 * The WP_Post object.
+	 * The WP_Post or WP_Comment object.
+	 *
+	 * This is the source object of the transformer.
 	 *
-	 * @var
+	 * @var WP_Post|WP_Comment
```
Comment (pfefferle) [inline includes/transformer/class-base.php:22]:
```suggestion
	 * @var stdClass|WP_Post|WP_Comment
```

File: includes/transformer/class-base.php (hunk 3 above)
Comment (Menrath) [reply includes/transformer/class-base.php:42]: Do you already have use cases in mind where a non WP_Post or WP_Comment class is passed? Just curious! Of course I am fine with accepting that suggestion.

I know that for example 'The Events Calendar' offers it's own object with handy functions, but one could always create an instance of that within the transformer by the WP_Post object.

Comment (pfefferle) [reply includes/transformer/class-base.php]: I fear that not all WordPress plugins work with custom post types, but I am not sure... You are the expert there... We can remove `stdClass`.

**Review by pfefferle — APPROVED** [review]
LGTM!

File: includes/transformer/class-base.php (hunk 3 above)
Comment (Menrath) [reply includes/transformer/class-base.php]: Some indeed use custom tables to handle complex functionality like recurring events, e.g. The Events Calendar. But that functionality is still built on top of a custom post type.

WordPress has been so clear for decades to use Custom Post Types wherever possible. I have not crossed any plugin that violated that suggestion.

**Review by pfefferle — APPROVED** [review]

**Pull request merged**